Assets/Scripts/Animation/GhostFadeAnim.cs

```diff
@@ -27,41 +29,62 @@ public class GhostFadeAnim : MonoBehaviour {
         if (spriteRenderer == null) {
             spriteRenderer = GetComponent<SpriteRenderer>();
             delayStart = System.DateTime.Now;
+			RandomizeFade ();
         }
     }
 
     void Update() {
         Color color = SpriteRenderer.color;
 
-        if (delayStart != default(System.DateTime) &&
-            (System.DateTime.Now - delayStart).TotalSeconds >= fadeDelay) {
-            delayStart = default(System.DateTime);
-        }
+		//original line: works at all
+		if (delayStart != default(System.DateTime) &&
+		(System.DateTime.Now - delayStart).TotalSeconds >= fadeDelay) {
+			//original line: works at all
+			delayStart = default(System.DateTime);
+		}
 
         if (fadeIn) {
             color.a += Time.deltaTime / fadeRate;
             if (color.a > 0.99f) {
-                if (dlgFade) { CompleteDialogueAction(); }
+				RandomizeFade ();
+                if (dlgFade) { 
+					CompleteDialogueAction(); 
+				}
                 delayStart = System.DateTime.Now;
                 fadeIn = false;
             }
-        }
-        else if (delayStart == default(System.DateTime)) {
+        } else if (delayStart == default(System.DateTime)) {
             color.a -= Time.deltaTime / fadeRate;
             if (color.a < 0.01f) {
-                if (dlgFade) { CompleteDialogueAction(); }
+                if (dlgFade) { 
+					CompleteDialogueAction(); 
+				}
                 fadeIn = true;
             }
         }
+
         SpriteRenderer.color = color;
     }
+		
+	//randomizes fade time and fade delay for ghosts before each fade
+	float RandomizeFade() {
+		float flt = Random.Range (2f, 5f);
```

Comment:
Let's base our range on fields that can be changed in the inspector. (minFadeRate, maxFadeRate)